docs/setup-guide.md

```diff
@@ -0,0 +1,47 @@
+(Setup Guide)=
+# Setup Guide <!-- omit in toc -->
+
+Essential information for MLRun developers and operators
+Learn about MLRun deployment options and version compatibility requirements.
+
+**In this section**
+- [MLRun client backward compatibility](#MLRun-client-backward-compatibility)
+- [Non root user support](#non-root-user-support)
+
+
+(MLRun-client-backward-compatibility)=
+## MLRun client backward compatibility  
+
+Starting from MLRun v1.3.0, the MLRun server is compatible with the client and images of the previous two minor MLRun releases. When you upgrade to v1.3.0, for example, you can continue to use your v1.1- and v1.2-based images, but v1.0-based images are not compatible.
+
+After you update the MLRun package client version by running `pip install mlrun==<"new-client-version">`, you must update the images to use the same client version you installed.
+For example, when running this command `pip install mlrun==1.8.0` you must update your images to use MLRun v1.8.0 by adding `mlrun==<"new-client-version">` as a function requirement. See {py:meth}`~mlrun.runtimes.BaseRuntime.with_requirements`.
+
+```{admonition} Important
+- Images from 0.9.0 are not compatible with 0.10.0. Backward compatibility starts from 0.10.0. 
+- When you upgrade the MLRun major version, for example 0.10.x to 1.0.x, there is no backward compatibility. 
+- The feature store is not backward compatible. 
+- When you upgrade the platform, for example from 3.2 to 3.3, the clients should be upgraded. There is no guaranteed compatibility with an older MLRun client after a platform upgrade. 
+```
+(Non-root-user)=
+## Non root user support
+
+By default, MLRun assigns the root user to MLRun runtimes and pods. You can improve the security context by changing the security mode, which is implemented by Iguazio during installation, and applied system-wide:
+
+- **Override**: Use the user id of the user that triggered the current run or use the nogroupid for group id. Requires Iguazio v3.5.1.
```

Comment:
- **Override**: Use the user id of the user that triggered the current run or use the nogroupid for group id. Requires Iguazio v3.5.1 and higher.